source/core/slang-shader-cache-index.cpp

```diff
@@ -0,0 +1,173 @@
+// slang-shader-cache-index.cpp
+#include "slang-shader-cache-index.h"
+
+#include "../slang/slang-hash-utils.h"
+#include "slang-io.h"
+#include "slang-string-util.h"
+#include "slang-file-system.h"
+
+namespace Slang
+{
+
+ShaderCacheIndex::ShaderCacheIndex(SlangInt size, String path, ISlangFileSystem* fileSystem, String filename)
+{
```

Comment:
Just rename this class to PersistentShaderCache since it also handle the actual shader code.